### DISCUSSION TITLE
How to restrict a user to create user below their access level?

### DISCUSSION THREAD

**Discussion by shahidcodes** [hasura/graphql-engine#9275] [Question]
Hi,

I have 5 roles in my system.

- Admin
- Manager
- Supervisor
- Team Head
- Staff

I want to restrict a user so that they can not create users with roles above them. i.e- The team head should create only Staff and the Supervisor should only create Team Head and Staff.

How can I achieve this?

**Comment by adas98012** [comment]
@shahidcodes 
You can assign numerical values to the roles hierarchy (e.g. Admin = 1, Manager = 2 etc...) and then define rules around it (e.g. Allow INSERT only if the User's Role Id < the role Id of the new User to be created). 

Please let me know if the info suffice.

**Reply by shahidcodes** [reply]
It'll suffice but I'm unable figure out how to get the request body in the permissions page? 

How do I actually write the permission? I get the access level from X-Hasura-Role but not the role user sent in graphql mutation.
